packages/console/src/pages/SignInExperience/PageContent/index.tsx

```diff
@@ -119,6 +130,41 @@ function PageContent({ data, onSignInExperienceUpdated }: Props) {
     }
   }, [isUploading, cancelUpload, reset]);
 
+  // Forgot password migration from null to normal array
+  useEffect(() => {
+    if (!isDevFeaturesEnabled) {
+      return;
+    }
+
+    // Wait for connectors list loading to be ready
+    if (!isConnectorsReady) {
+      return;
+    }
+
+    // If there is no password method, we should clear the forgot password methods.
+    if (!hasPasswordMethod) {
+      setValue('forgotPasswordMethods', []);
+    } else if (!formData.forgotPasswordMethods) {
+      // If this is null, we should initialize it based on current connector setup
+      // if has email connector, then add email verification code method, also for sms connector
+      const initialMethods = [
+        ...(isConnectorTypeEnabled(ConnectorType.Email)
+          ? [ForgotPasswordMethod.EmailVerificationCode]
+          : []),
+        ...(isConnectorTypeEnabled(ConnectorType.Sms)
+          ? [ForgotPasswordMethod.PhoneVerificationCode]
+          : []),
+      ];
+
+      setValue('forgotPasswordMethods', initialMethods, { shouldDirty: true });
+      setIsForgotPasswordMigrationNoticeVisible(true);
+      void onSubmit({
+        ...formData,
+        forgotPasswordMethods: initialMethods,
+      });
```

Comment:
Why should we auto submit the change? The submit action has to be trigger manually by the user, right?